ee/session_recordings/session_recording_playlist.py

```diff
@@ -224,10 +230,19 @@ def recordings(self, request: request.Request, *args: Any, **kwargs: Any) -> res
             .values_list("recording_id", flat=True)
         )
 
-        filter = SessionRecordingsFilter(request=request, team=self.team)
-        filter = filter.shallow_clone({SESSION_RECORDINGS_FILTER_IDS: json.dumps(playlist_items)})
+        use_query_type = (request.GET.get("as_query", "False")).lower() == "true"
 
-        return list_recordings_response(filter, request, self.get_serializer_context())
+        if use_query_type:
+            data_dict = query_as_params_to_dict(request.GET.dict())
+            query = RecordingsQuery.model_validate(data_dict)
+            query.session_ids = playlist_items
```

Comment:
Are we only doing this validation because it doesn't go through the `/query` endpoint? Maybe we should just move it to a runner instead. Totally happy if that's the next step in a followup PR